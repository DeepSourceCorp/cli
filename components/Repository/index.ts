import GistCard from './GistCard.vue'
import GistCardTitle from './GistCardTitle.vue'
import GistCardDescription from './GistCardDescription.vue'
import IssueType from './IssueType.vue'
import IgnoredRule from './IgnoredRule.vue'
import IssueSeverityTag from './IssueSeverityTag.vue'

export { GistCard, GistCardTitle, GistCardDescription, IssueType, IgnoredRule, IssueSeverityTag }
