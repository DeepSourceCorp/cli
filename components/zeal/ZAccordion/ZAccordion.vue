<template>
  <div>
    <slot></slot>
  </div>
</template>

<script>
export default {
  name: 'ZAccordion',
  components: {},
  props: {
    defaultStyle: {
      type: Boolean,
      default: false
    },
    disabled: {
      type: Boolean,
      default: false
    },
    allowMultiple: {
      type: Boolean,
      default: true
    },
    showBorders: {
      type: Boolean,
      default: false
    }
  },
  data() {
    return {
      accordionItems: []
    }
  },
  computed: {}
}
</script>
