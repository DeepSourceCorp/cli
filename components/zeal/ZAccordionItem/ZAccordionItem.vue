<template>
  <div
    class="z-accordion-item"
    :class="{
      'border-b border-t border-ink-200 first:border-0 last:border-0 hover:bg-ink-300': showBorders,
      'text-vanilla-200': !isDisabled,
      'text-slate': isDisabled
    }"
  >
    <slot name="title" :open="open" :toggle-accordion="toggleAccordion">
      <div
        class="group flex items-center transition-all duration-700 ease-in-out"
        :class="{
          'cursor-not-allowed': isDisabled,
          'cursor-pointer': !isDisabled && !isList
        }"
        @click="toggleAccordion()"
      >
        <span class="flex-1 font-medium">{{ title }}</span>
        <z-icon
          icon="chevron-right"
          color="slate"
          class="duration-DEFAULT group transform stroke-1.5 transition-all ease-in-out"
          size="medium"
          :class="[
            accordionHeaderAnimations,
            {
              'rotate-90 text-vanilla-200': open,
              'text-slate': isDisabled,
              'group-hover:text-vanilla-200': !isDisabled,
              hidden: isList
            }
          ]"
        />
      </div>
    </slot>
    <div
      class="overflow-hidden text-sm leading-6 transition-max-height duration-300 ease-in-out"
      :class="[
        open ? (spanCustomHeight ? customMaxHeight : 'max-h-52') : 'max-h-0',
        { 'max-h-full': isList }
      ]"
    >
      <slot></slot>
    </div>
  </div>
</template>

<script>
export default {
  name: 'ZAccordionItem',
  props: {
    title: {
      default: '',
      type: String
    },
    isOpen: {
      type: Boolean,
      default: false
    },
    icon: {
      type: String,
      default: 'down'
    },
    isList: {
      type: Boolean,
      default: false
    },
    spanCustomHeight: {
      type: Boolean,
      default: false
    },
    customMaxHeight: {
      type: String,
      default: 'max-h-screen'
    }
  },
  data() {
    return {
      open: this.isOpen,
      accordionHeaderAnimations: '',
      isDisabled: this.$parent.disabled,
      showBorders: this.$parent.showBorders
    }
  },
  watch: {
    isOpen() {
      this.toggleAccordion()
    }
  },
  methods: {
    toggleAccordion() {
      if (this.isDisabled) return
      if (this.isList) return
      this.open = !this.open
      this.accordionHeaderAnimations = this.open
        ? 'animate-first-quarter-spin rotate-90'
        : 'animate-reverse-quarter-spin rotate-0'
      this.$emit('isOpen', this.open)
    }
  }
}
</script>
