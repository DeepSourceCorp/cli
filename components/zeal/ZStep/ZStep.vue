<template>
  <div
    class="z-step mr-0"
    :class="{
      'w-1/3 flex-shrink-0 flex-grow-0': align === 'horizontal' && isLast,
      'w-1/2': align === 'horizontal' && !isLast,
      'flex flex-row': align === 'vertical'
    }"
  >
    <div
      class="z-step__head relative"
      :class="{
        currentStatus: true,
        'w-full': align === 'horizontal',
        'flex flex-col justify-center': align === 'vertical' && !isLast
      }"
    >
      <div
        v-if="align === 'horizontal' && !isLast"
        class="z-step__line absolute left-7 right-1 top-2.5 h-0.5 w-auto rounded-sm"
        :class="{
          'bg-juniper': currentStatus === 'completed',
          'bg-ink-200': currentStatus !== 'completed'
        }"
      ></div>
      <div
        class="z-step__icon h-6 w-6 rounded-full"
        :class="{
          'bg-juniper': currentStatus === 'completed',
          'bg-robin': currentStatus === 'active',
          'bg-ink-200': !showNumbers && currentStatus === 'default',
          'bg-ink-100': showNumbers && currentStatus === 'default',
          'flex-shrink-0': align === 'vertical'
        }"
      >
        <span
          v-if="showNumbers"
          class="flex h-full items-center justify-center text-xs font-medium leading-none text-vanilla-100"
        >
          {{ index + 1 }}
        </span>
        <span
          v-else-if="currentStatus === 'active'"
          class="flex h-full items-center justify-center"
        >
          <z-icon icon="circle-dashed" color="vanilla-100" class="stroke-2" />
        </span>
        <span
          v-else-if="currentStatus === 'completed'"
          class="flex h-full items-center justify-center"
        >
          <z-icon icon="check" color="ink-400" class="stroke-2" />
        </span>
      </div>
      <div
        v-if="align === 'vertical' && !isLast"
        class="z-step__line mx-auto my-1 h-full w-0.5 rounded-sm"
        :class="{
          'bg-juniper': currentStatus === 'completed',
          'bg-ink-200': currentStatus !== 'completed'
        }"
      ></div>
    </div>
    <div
      class="z-step__main"
      :class="{ 'mt-4': align === 'horizontal', 'mb-4 ml-4 w-full': align === 'vertical' }"
    >
      <slot name="title">
        <div
          class="z-step__title mt-1 text-xs font-medium uppercase leading-snug tracking-widest text-vanilla-400"
        >
          {{ title }}
        </div>
      </slot>
      <div
        class="z-step__description mt-1 text-sm font-bold leading-6"
        :class="{
          'text-vanilla-200': currentStatus,
          'text-vanilla-400': currentStatus === 'default'
        }"
      >
        <slot name="description" :status="currentStatus">{{ description }}</slot>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'ZStep',
  props: {
    title: {
      required: false,
      default: undefined,
      type: String
    },
    description: {
      required: false,
      default: undefined,
      type: String
    },
    status: {
      required: false,
      type: String,
      default: 'default',
      validator: (value) => {
        return ['default', 'active', 'completed'].includes(value)
      }
    }
  },
  data() {
    return {
      index: -1,
      showNumbers: false,
      internalStatus: 'default',
      align: 'horizontal'
    }
  },
  computed: {
    currentStatus() {
      return this.status || this.internalStatus
    },
    isLast() {
      const parent = this.$parent
      return parent.steps[parent.steps.length - 1] === this
    }
  },
  beforeCreate() {
    const parent = this.$parent
    parent.steps.push(this)
  },
  beforeDestroy() {
    const parent = this.$parent,
      steps = parent.steps
    const index = steps.indexOf(this)
    if (index >= 0) {
      steps.slice(index, 1)
    }
  }
}
</script>
