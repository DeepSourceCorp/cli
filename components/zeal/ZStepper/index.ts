import ZStepper from './ZStepper.vue'

export default ZStepper
