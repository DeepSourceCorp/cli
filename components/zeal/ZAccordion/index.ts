import ZAccordion from './ZAccordion.vue'

export default ZAccordion
