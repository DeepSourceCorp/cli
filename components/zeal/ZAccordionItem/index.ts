import ZAccordionItem from './ZAccordionItem.vue'

export default ZAccordionItem
