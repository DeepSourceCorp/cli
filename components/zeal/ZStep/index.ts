import ZStep from './ZStep.vue'

export default ZStep
