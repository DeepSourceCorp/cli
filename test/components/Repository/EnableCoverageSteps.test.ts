import { render } from '@testing-library/vue'
import ZAccordion from '@/components/zeal/ZAccordion'
import ZAccordionItem from '@/components/zeal/ZAccordionItem'
import ZStepper from '@/components/zeal/ZStepper'
import ZStep from '@/components/zeal/ZStep'

import { EnableCoverageSteps } from '~/components/Repository'

test('renders EnableCoverageSteps with all prop options', () => {
  const props = { dsn: 'http://6f7d439568e64460a5e0a31967314288@localhost:3000' }

  const { html } = render(EnableCoverageSteps, {
    props,
    stubs: {
      ZAccordion,
      ZAccordionItem,
      ZStepper,
      ZStep,
      ZIcon: true,
      ZSelect: true,
      ZOption: true,
      unstyledCopyButton: true
    }
  })

  expect(html()).toMatchSnapshot(JSON.stringify(props))
})
