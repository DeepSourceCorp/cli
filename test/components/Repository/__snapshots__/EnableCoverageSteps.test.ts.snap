// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders EnableCoverageSteps with all prop options: {"dsn":"http://6f7d439568e64460a5e0a31967314288@localhost:3000"} 1`] = `
"<div class=\\"p-4 pb-0\\">
  <div class=\\"z-accordion-item text-vanilla-200\\">
    <div class=\\"flex cursor-pointer items-center gap-1 pb-4\\">
      <zicon-stub icon=\\"chevron-right\\" size=\\"x-small\\" class=\\"transform\\"></zicon-stub> <span class=\\"text-xs leading-6 text-vanilla-400\\">
          How to send coverage reports to DeepSource?
        </span>
    </div>
    <div class=\\"overflow-hidden text-sm leading-6 transition-max-height duration-300 ease-in-out max-h-0\\">
      <div class=\\"z-stepper flex flex-col\\">
        <div class=\\"z-step mr-0 w-1/3 flex-shrink-0 flex-grow-0\\">
          <div class=\\"z-step__head relative currentStatus w-full\\">
            <!---->
            <div class=\\"z-step__icon h-6 w-6 rounded-full bg-ink-200\\">
              <!---->
            </div>
            <!---->
          </div>
          <div class=\\"z-step__main mt-4\\">
            <p class=\\"text-xs text-vanilla-400\\">
              Set up the
              <code class=\\"bifrost-inline-code uppercase text-vanilla-100\\">DEEPSOURCE_DSN</code>
              environment variable.
            </p>
            <div class=\\"z-step__description mt-1 text-sm font-bold leading-6 text-vanilla-200 text-vanilla-400\\">
              <div class=\\"copy-content-container h-8 max-w-2xs items-center gap-x-4 py-2 sm:max-w-none\\"><code class=\\"truncate font-mono text-xs font-normal leading-4 text-vanilla-100\\">
              http://6f7d439568e64460a5e0a31967314288@localhost:3000
            </code>
                <unstyledcopybutton-stub value=\\"http://6f7d439568e64460a5e0a31967314288@localhost:3000\\" icon-size=\\"x-small\\" class=\\"auxillary-button text-xs text-vanilla-100\\"></unstyledcopybutton-stub>
              </div>
            </div>
          </div>
        </div>
        <div class=\\"z-step mr-0 w-1/3 flex-shrink-0 flex-grow-0\\">
          <div class=\\"z-step__head relative currentStatus w-full\\">
            <!---->
            <div class=\\"z-step__icon h-6 w-6 rounded-full bg-ink-200\\">
              <!---->
            </div>
            <!---->
          </div>
          <div class=\\"z-step__main mt-4\\">
            <p class=\\"mb-2 text-xs text-vanilla-400\\">
              Install the
              <a href=\\"https://deepsource.com/cli\\" target=\\"blank\\" rel=\\"noreferrer noopener\\" class=\\"inline-flex items-center gap-x-1 text-juniper\\">
                DeepSource CLI
                <zicon-stub icon=\\"arrow-up-right\\" size=\\"x-small\\" color=\\"current\\"></zicon-stub></a></p>
            <div class=\\"z-step__description mt-1 text-sm font-bold leading-6 text-vanilla-200 text-vanilla-400\\">
              <div class=\\"flex max-w-2xs items-center gap-2 font-normal sm:max-w-none\\">
                <div class=\\"h-8 w-24 flex-shrink-0\\">
                  <zselect-stub border-radius=\\"rounded-sm\\" spacing=\\"p-1.5\\" value=\\"curl\\">
                    <zoption-stub label=\\"cURL\\" value=\\"curl\\" class=\\"z-20\\"></zoption-stub>
                    <zoption-stub label=\\"Homebrew\\" value=\\"brew\\" class=\\"z-20\\"></zoption-stub>
                  </zselect-stub>
                </div>
                <div class=\\"copy-content-container h-8 min-w-0 flex-grow items-center gap-x-2 py-2 sm:min-w-max\\"><code class=\\"truncate font-mono text-xs leading-3 text-vanilla-100\\"><span class=\\"select-none text-honey\\">$</span> 
                  curl
                  <span class=\\"text-aqua-500\\">deepsource.io/cli</span> <span class=\\"text-honey\\"> | sh</span></code>
                  <unstyledcopybutton-stub value=\\"curl deepsource.io/cli | sh\\" label=\\"Copy\\" icon-size=\\"x-small\\" class=\\"auxillary-button text-xs text-vanilla-100\\"></unstyledcopybutton-stub>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class=\\"z-step mr-0 w-1/3 flex-shrink-0 flex-grow-0\\">
          <div class=\\"z-step__head relative currentStatus w-full\\">
            <!---->
            <div class=\\"z-step__icon h-6 w-6 rounded-full bg-ink-200\\">
              <!---->
            </div>
            <!---->
          </div>
          <div class=\\"z-step__main mt-4\\">
            <p class=\\"mb-2 text-xs leading-6 text-vanilla-400\\">
              Report your coverage artifact to DeepSource using the CLI.
            </p>
            <div class=\\"z-step__description mt-1 text-sm font-bold leading-6 text-vanilla-200 text-vanilla-400\\">
              <div class=\\"copy-content-container max-w-2xs items-start sm:max-w-none\\"><code class=\\"line-clamp-3 max-w-md py-2 font-mono text-xs font-normal leading-4 text-vanilla-100\\">
              ./bin/deepsource report --analyzer test-coverage --key &lt;key&gt; --value-file &lt;absolute-path-of-the-coverage report&gt;
            </code>
                <unstyledcopybutton-stub value=\\"./bin/deepsource report --analyzer test-coverage --key <key> --value-file <absolute-path-of-the-coverage report>\\" icon-size=\\"x-small\\" class=\\"auxillary-button mt-1 text-xs text-vanilla-100\\"></unstyledcopybutton-stub>
              </div> <a href=\\"https://docs.deepsource.com/docs/analyzers-test-coverage#setup-test-coverage\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" class=\\"auxillary-link mt-3 h-7 w-max text-xs font-normal text-vanilla-400\\">
                Read full documentation
                <zicon-stub icon=\\"arrow-up-right\\" size=\\"x-small\\" color=\\"current\\"></zicon-stub></a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
